kitsune/wiki/facets.py

```diff
@@ -108,12 +108,14 @@ def _documents_for(user, locale, topics=None, products=None):
     # speed up query by removing any ordering, since we're doing it in python:
     qs = qs.select_related("current_revision", "parent").order_by()
 
+    topic_ids = [t.id for t in topics]
     for topic in topics or []:
         # we need to filter against parent topics for localized articles
-        qs = qs.filter(Q(topics=topic) | Q(parent__topics=topic))
+        qs = qs.filter(Q(topics__in=topic_ids) | Q(parent__topics__in=topic_ids))
```

Comment:
With this approach, we don't need the loop, and since `topics` can be `None`, maybe something like the following instead?

```python
    if topics:
        topic_ids = [t.id for t in topics]
        # we need to filter against parent topics for localized articles
        qs = qs.filter(Q(topics__in=topic_ids) | Q(parent__topics__in=topic_ids))
```